### PR TITLE
Documentation Improvements

### PR DESCRIPTION
**Pull request by apahim**
- Organize release notes.
 - Create captions so the left menu can have sections.
 - Improve index.

### CODE REVIEW

**Comment by apahim** [comment]
@clebergnu @ldoktor you can see the result here:
http://apahim-avocado.readthedocs.io/en/improve_docs/

File: docs/source/release_notes/0_32_0.rst [hunk 1]
```diff
@@ -1,13 +1,10 @@
-===================
-0.32.0: Road Runner
-===================
+====================
+0.32.0 (Road Runner)
```
Comment (ldoktor) [inline docs/source/release_notes/0_32_0.rst]: Honestly I'd go with `0.32.0 Road Runner`, `0.32.0 - Road Runner`, `0.32.0: Road Runner` or `0.32.0 AKA Road Runner`. I don't like the brackets, though.

File: docs/source/index.rst [hunk 2]
```diff
@@ -6,6 +6,7 @@ Contents:
 
 .. toctree::
    :maxdepth: 2
+   :caption: Avocado User's Guide
```
Comment (ldoktor) [inline docs/source/index.rst]: this renders well, I like it.

File: docs/source/release_notes/36_0.rst [hunk 3]
```diff
@@ -1,12 +1,12 @@
 =======
-36.0lts
+36.0LTS
```
Comment (apahim) [inline docs/source/release_notes/36_0.rst]: 36.0lts, 36.0LTS or 36.0 LTS?

**Comment by ldoktor** [comment]
I like this. Basically the only complains I have are:

* release notes index page, I'd prefer the version and names to be aligned, if possible (both as links to the details)
* The version+name label - https://github.com/avocado-framework/avocado/pull/1317/commits/01998a014fb5d40b30dd3859bb86421c183c97ce#r71730027

Apart from that it looks really nice.

File: docs/source/release_notes/36_0.rst (hunk 3 above)
Comment (ldoktor) [inline docs/source/release_notes/36_0.rst]: The official name was `36.0lts` but that is incorrect version according to PEP. So new LTS releases will be just the version and the `LTS` would only be present in the name: https://trello.com/c/WI1Nwpdb/708-evaluate-version-numbering-scheme-for-lts-releases

So if I understand the result correctly, it should be `36.0 - LTS` (or `36.0 (LTS)` but I really hate the format of that)

**Pull request closed without merging**